tools/crm_resource.c

```diff
@@ -39,28 +39,34 @@ static pe_working_set_t *data_set = NULL;
 #define MESSAGE_TIMEOUT_S 60
 
 // Clean up and exit
-static void
+static crm_exit_t
 bye(crm_exit_t exit_code)
 {
-    static bool crm_resourece_shutdown_flag = FALSE;
+    static bool crm_resource_shutdown_flag = false;
 
-    if (crm_resourece_shutdown_flag) {
-        return;
+    if (crm_resource_shutdown_flag) {
+        // Allow usage like "return bye(...);"
+        return exit_code;
     }
-    crm_resourece_shutdown_flag = TRUE;
+    crm_resource_shutdown_flag = true;
 
     if (cib_conn != NULL) {
-        cib_conn->cmds->signoff(cib_conn);
-        cib_delete(cib_conn);
+        cib_t *save_cib_conn = cib_conn;
+
         cib_conn = NULL;
+        save_cib_conn->cmds->signoff(save_cib_conn);
+        cib_delete(save_cib_conn);
     }
     if (controld_api != NULL) {
-        pcmk_free_controld_api(controld_api);
+        pcmk_controld_api_t *save_controld_api = controld_api;
+
+        controld_api = NULL;
+        pcmk_free_controld_api(save_controld_api);
     }
     pe_free_working_set(data_set);
     data_set = NULL;
     crm_exit(exit_code);
-    return;
+    return exit_code;
```

Comment:
An alternate might be to mark this function as _Noreturn.  We do that in a couple places already.  It doesn't really matter to me one way or the other.

Reply:
Except it's not now :) which is the point (it can return at the beginning)